### PR TITLE
Fix project_normal in GlobeView

### PR DESCRIPTION
**Pull request by Pessimistress**
For #6423

This PR makes `project_normal` depend on `geomotry.position` in globe projection mode. The layers' vertex shader statements are rearranged so that normals are calculated after position. If a shader happens to call `project_normal` before populating `geometry.position`, the rotation matrix comes out as an identity mat3, i.e. no change from the existing behavior.

#### Change List
- `project_normal` function rotates normal to align with local up in globe projection mode
- rearrange 3D layers' vertex shaders so that normals are calculated after position
- Update render test

### CODE REVIEW

**Review by ibgreen — APPROVED** [review]
This looks great, should enable improved visual effects in globe mode!

I will defer to @igorDykhta 's eagle eyes for detailed shader review.

**Comment by ibgreen** [comment]
You are getting the same bizarre test failures as I am doing in my open PRs. For some reason I was able to make one line change PRs without triggering these, but as soon as I change a bunch of files I get these 19 failures (including the flaky 33 -> 34 picking issue).

File: modules/geo-layers/src/mesh-layer/mesh-layer-vertex.glsl.js [hunk 1]
```diff
@@ -30,46 +30,40 @@ void main(void) {
...
-  geometry.normal = normals_commonspace;
 
   vec3 pos = (instanceModelMatrix * positions) * sizeScale;
   vec3 projectedPosition = project_position(positions);
   position_commonspace = vec4(projectedPosition, 1.0);
   gl_Position = project_common_position_to_clipspace(position_commonspace);
 
+  normals_commonspace = project_normal(instanceModelMatrix * normals);
+  geometry.normal = normals_commonspace;
   geometry.position = position_commonspace;
```
Comment (igorDykhta) [inline modules/geo-layers/src/mesh-layer/mesh-layer-vertex.glsl.js:50]: Should `geometry.position = position_commonspace;` be before `project_normal`?

**Review by igorDykhta — APPROVED** [review]

**Comment by ibgreen** [comment]
@Pessimistress fixed CI by reverting yarn.lock changes. Good to land!

**Pull request merged**